### PR TITLE
Add karma

### PR DESCRIPTION
**Pull request by k88hudson**
This uses Chrome as a runner right now pending https://github.com/mozilla/fathom/pull/25. I'm not as familiar with circle ci but we'll probably have to do something to get karma to run (probs on FF)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,10 +4,11 @@
   "description": "A Javascript library for parsing metadata in a Web Page.",
   "main": "parser.js",
   "scripts": {
-    "test:lint": "./node_modules/eslint/bin/eslint.js ./*.js ./tests/*.js",
-    "test:mocha": "./node_modules/istanbul/lib/cli.js cover _mocha --report lcovonly -- tests/* -R spec",
-    "test": "./node_modules/npm-run-all/bin/npm-run-all/index.js test:*",
-    "cover": "cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js"
+    "test:lint": "eslint ./*.js ./tests/*.js",
```
Comment (pdehaan) [inline package.json]: :+1: 
You could probably optimize this to just `"eslint ."` (and we can use .eslintignore to ignore any files we may not care about).

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    preprocessors: {
+      'tests/**/*.js': ['webpack', 'sourcemap']
+    },
+    webpack: {
+      module: {
+        loaders: [{test: /\.json$/, loader: 'json'}],
+        postLoaders: [{
+          test: /\.js$/,
+          loader: 'istanbul-instrumenter',
+          include: [path.join(__dirname, '/parser.js')]
```
Comment (pdehaan) [inline karma.conf.js]: probably remove the leading "/" in "/parser.js" here (since `path.join()` should infer the directory separator based on OS).

**Comment by pdehaan** [comment]
Not 💯 on why this failed on Circle-CI, but this **Firefox 38.0.0** reference looks _mighty_ suspicious:

```
INFO [Firefox 38.0.0 (Linux 0.0.0)]: Connected on socket /#t_Bv-c8J_6HJ6sW-AAAA with id 81507131
Firefox 38.0.0 (Linux 0.0.0) ERROR
  SyntaxError: missing ; before statement
  at webpack:///parser.js:9:0 <- tests/index.js:8207

```

**Comment by jaredkerim** [comment]
Closing this to be replaced by a more comprehensive pull request.

**Pull request closed without merging**